code/modules/projectiles/gun.dm

```diff
@@ -126,7 +125,17 @@
 				user.visible_message("<span class='danger'>[user] fires [src]!</span>", null, null, COMBAT_MESSAGE_RANGE)
 
 
+//Adds logging to the attack log whenever anyone draws a gun, adds a pause after drawing a gun before you can do anything based on it's size
+/obj/item/gun/pickup(mob/living/user)
+	. = ..()
+	user.visible_message("<span class='danger'>[user] grabs a gun!</span>") // probably could code in differences as to where you're picking it up from and so forth. later.
+	user.Paralyze(2 * src.weapon_weight, ignore_canparalyze = TRUE) // Need to define where you're grabbing it from, assign numbers to them, and then divide the paralyze total by that. Tables/holster/belt/back/container.
+	user.log_message("[user] drew a gun", INDIVIDUAL_ATTACK_LOG)
+	src.slowdown = (w_class / 2)
```

Comment:
note to self, bind this all into a proc under obj/item
weapon_weight might be a problem though

Reply:
fuck you past me I thought I was done